lib/malan_web/controllers/session_controller.ex

```diff
@@ -145,6 +146,68 @@ defmodule MalanWeb.SessionController do
 
   def show_current(conn, %{}), do: show(conn, %{"id" => conn.assigns.authed_session_id})
 
+  def extend(conn, attrs = %{"id" => id}) do
+    session = Accounts.get_session!(id)
+
+    # Make sure the session being extended isn't revoked or expired
+    case Accounts.session_revoked_or_expired?(session) do
+      true ->
+        conn
+        |> put_status(403)
+        |> put_view(ErrorView)
+        |> render("403.json", session_revoked_or_expired: true)
+
+      false ->
+        extend_session(conn, session, attrs)
+    end
+  end
+
+  def extend_current(conn, attrs),
+    do: extend(conn, Map.merge(attrs, %{"id" => conn.assigns.authed_session_id}))
+
+  defp extend_session(conn, session, attrs) do
+    authed_ids = %{
+      authed_user_id: conn.assigns.authed_user_id,
+      authed_session_id: conn.assigns.authed_session_id
+    }
+
+    changeset =
+      Session.extend_changeset(
+        session,
+        %{
+          extend_by_seconds: attrs["extend_by_seconds"]
```

Comment:
on my first read, i thought this would add a number of seconds to the current expiration. however according to https://github.com/FreedomBen/malan/pull/150/files#diff-ce0c8a40540c29dc0b3c9e86761573841e1ee1845b2effebef4bd8fea90e896dR273 it looks like it may only extend to a number of seconds _from now_.

maybe another param name could avoid that confusion? not sure if there's already a convention in play, but `expire_in_seconds` might get the point across.